CMakeLists.txt

```diff
@@ -225,11 +225,14 @@ if (ENABLE_EXPENSIVE_CHECKS)
     add_definitions("-DAD_ENABLE_EXPENSIVE_CHECKS")
 endif()
 
-if (DISABLE_QUERY_CANCELLATION_WATCH_DOG)
-    message(STATUS "Disable periodic checks that the cancellation flag is checked frequently enough while the result of
-    a query is being computed with a small runtime overhead")
-    add_definitions("-DQLEVER_DISABLE_QUERY_CANCELLATION_WATCH_DOG")
+set(QUERY_CANCELLATION_MODE "ENABLED" CACHE STRING "Option to allow disabling cancellation checks partially or completely to reduce the overhead of this mechanism during query computation.")
+# Hint for cmake gui, but not actually enforced
+set_property(CACHE QUERY_CANCELLATION_MODE PROPERTY STRINGS "ENABLED" "NO_WATCH_DOG" "DISABLED")
+# So enforce this ourselves
+if(QUERY_CANCELLATION_MODE AND NOT QUERY_CANCELLATION_MODE MATCHES "ENABLED|NO_WATCH_DOG|DISABLED")
+    message(FATAL_ERROR "Invalid value for QUERY_CANCELLATION_MODE '${QUERY_CANCELLATION_MODE}'. Please remove the option entirely or change it to ENABLED, NO_WATCH_DOG or DISABLED.")
```

Comment:
Can you try turning this into a `CMake` macro s.t. we just can write something like`
`ad_add_property(CANCELLATION_MODE "Description..." "ValidOption1" "ValidOption2" ...)`
(maybe also for another PR).

Reply:
This could then immediately be reused for the Loglevels which have exactly the same stuff.
And maybe then we can also set the various bools true/false to proper `ON/OFF` properties, which seem to be the standard for CMake.
(Okay, maybe this IS another PR:)